homeassistant/components/androidtv/media_player.py

```diff
@@ -414,7 +418,10 @@ def __init__(self, aftv, name, apps, turn_on_command, turn_off_command):
         self._device = None
         self._device_properties = self.aftv.device_properties
         self._is_volume_muted = None
-        self._unique_id = self._device_properties.get("serialno")
+        self._unique_id = None
+        serialno = self._device_properties.get("serialno")
```

Comment:
If serialno is empty, what does this variable contain? Does it mean that the key is there but it holds an empty string?

Reply:
by the former implementation, the variable will contain an string which length is zero. And it will leads to the entities registry conflicts.

Reply:
I would change this to

```python
if serialno and serialno.strip():
    self._unique_id = serialno.strip()
```

This avoids the case that `serialno` is just a space.

Reply:
Also, you should revert all other changes, since this is the only real change to the code.